server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferMetadata.java

```diff
@@ -122,6 +142,25 @@ private void write(DataOutput out) throws IOException {
         out.writeLong(generation);
         out.writeLong(minTranslogGeneration);
         out.writeLong(timeStamp);
-        out.writeMapOfStrings(generationToPrimaryTermMapper.get());
+        if (generationToPrimaryTermMapper.get() != null) {
+            out.writeMapOfStrings(generationToPrimaryTermMapper.get());
+        } else {
+            out.writeMapOfStrings(new HashMap<>());
+        }
+    }
+
+    private static class MetadataFilenameComparator implements Comparator<String> {
+        @Override
+        public int compare(String metadaFilename1, String metadaFilename2) {
+            // Format of metadata filename is <Primary Term>__<Generation>__<Timestamp>
+            String[] filenameTokens1 = metadaFilename1.split(METADATA_SEPARATOR);
+            String[] filenameTokens2 = metadaFilename2.split(METADATA_SEPARATOR);
+            for (int i = 0; i < filenameTokens1.length; i++) {
+                if (filenameTokens1[i].equals(filenameTokens2[i]) == false) {
+                    return (int) (Long.parseLong(filenameTokens1[i]) - Long.parseLong(filenameTokens2[i]));
```

Comment:
Maybe do `return Long.compare(Long.parseLong(filenameTokens1[i]), Long.parseLong(filenameTokens2[i]))` instead?

If somehow a non-conforming filename is present this will fail with either a NumberFormatException or an ArrayIndexOutOfBoundsException. Do we need to defend against those cases?